src/common/utils/ldap/ldap.go

```diff
@@ -401,6 +401,21 @@ func (session *Session) searchGroup(baseDN, filter, groupName, groupNameAttribut
 	return ldapGroups, nil
 }
 
+// SearchGroupNameByDN - search ldap group name by DN
+func (session *Session) SearchGroupNameByDN(groupDN string) (string, error) {
+	if len(session.ldapGroupConfig.LdapGroupNameAttribute) == 0 {
+		return groupDN, nil
+	}
+	groups, err := session.SearchGroupByDN(groupDN)
+	if err != nil {
+		return "", err
+	}
+	if len(groups) == 0 {
+		return "", fmt.Errorf("failed to retrieve group by DN: %v", groupDN)
+	}
+	return groups[0].GroupName, nil
```

Comment:
How does `LdapGroupNameAttribute` take effect here?

Reply:
It is used in https://github.com/goharbor/harbor/blob/3c345f92fd49c0300f6a54b3b1bf3c99a5cd7bc6/src/common/utils/ldap/ldap.go#L368

Reply:
But in `SearchGroupByDN` you've already populated the groupname into each entry?  It feels like there are redundant funcs in this file.

Reply:
Remove the SearchGroupNameByDN